core/src/main/java/org/apache/calcite/rel/metadata/RelMdRowCount.java

```diff
@@ -169,6 +170,12 @@ public Double getRowCount(Calc rel, RelMetadataQuery mq) {
     return limit < rowCount ? limit : rowCount;
   }
 
+  public @Nullable Double getRowCount(Sample rel, RelMetadataQuery mq) {
```

Comment:
This is make sense for me.

Reply:
Yes. Perhaps worth justifying with a comment (because nothing is obvious in statistics). The expected value of a binomial distribution with probability p over n trials is n * p.